Create Vomit Puddle and Spawn Skeleton From Puddle

Lukas-bit opened the pull request:
The behemoth vole's projectile attack will now turn into a poison "puddle" at the end of its lifespan. There will also be a "once swallowed man" - who is now a skeleton - that will come out of the vomit puddle to chase after and attack the player. Also, an important note is that I changed the player and other enemy types to be rendered at a Z index of 1. This lets me reduce the Z index of the vomit puddle, making it render behind the player and enemies.

Lukas-bit committed:
Merge branch 'master' of https://github.com/noahwagner04/CSE320_Game

Lukas-bit committed:
Merge branch 'master' into lukas/behemoth_projectile_attack

Lukas-bit committed:
Create Poison Puddle At End of Projectile Lifetime

Once the vomit projectile that the boss shoots has reached the end of its lifetime,
a new sprite will replace the projectile one indicating the it is now a vomit puddle.
This puddle is poisonous. Also, I changed the visibility layer of all enemies to
be top level, along with the initial poison projectile. This way, once the projectile turns
into a puddle, I can change the visibility layer of the puddle to be behind every enemy.

Lukas-bit committed:
Add Secondary Enemy Spawn to Projectile Attack

A skeleton warrior who was "swallowed" by the behemoth vole will now
spawn out of the puddles of vomit left behind from the boss's projectile
attack and will chase after the player.

Lukas-bit committed:
Merge branch 'lukas/behemoth_projectile_attack' of https://github.com/noahwagner04/CSE320_Game into lukas/behemoth_projectile_attack

Lukas-bit committed:
Merge branch 'master' of https://github.com/noahwagner04/CSE320_Game

Lukas-bit committed:
Merge branch 'master' into lukas/behemoth_projectile_attack

Lukas-bit committed:
Connect Health Container to Skeleton

Forgot to connect the health container to the skeletons. Also, I increased
the knockback dealt to them and decreased the max number of vomit attacks

Lukas-bit committed:
Create Second HitBox Node for Sword

There's now a separate hitbox for the swallowed man's sword that deals
more damage to the player compared to the body hitbox.

Lukas-bit committed:
Set Top-Level False and Increase Z Index Instead

Instead of having the player and enemies be on the top level compared to
other canvas items, change their render order by using the Z index.

obimax99 reviewed (changes requested):
vomit_projectile.gd has a line referencing %Sprite2D which I'm pretty sure means there's one sprite2d that has a Unique Special name of Sprite2D and that's not great. If it does need to be unique for some reason then we should change the name to something like %VomitProjectileSprite, if it doesn't need to be unique then we can just change the code to use $Sprite2D

Lukas-bit committed:
Access Child Nodes By Relative Path

Changed how I access the child nodes of the behemoth vole, vomit
projectile, and swallowed man. They will now be accessed by their
relative path to the calling node instead of by their names.

obimax99 reviewed (approved):
Looks good!
